### PR TITLE
Neo4J-update-t-4.1

### PR DESCRIPTION
**Pull request by MattGalvis**

### CODE REVIEW

**Review by trentmurray — APPROVED** [review]
Changes look sane.

**Comment by trentmurray** [comment]
@robinedwards  - any chance we can get a newer version out?

**Review by robinedwards — COMMENTED** [review]
I will take a look now

**Comment by MattGalvis** [comment]
I specifically put this here to figure out the tests Im still trying to figure out. any help would be awesome

**Comment by robinedwards** [comment]
It seems the credentials are wrong when connecting to the database i can see an AuthError i the trace. On import of tests/__init__.py it tries to clear the database (probably not a sensible idea of mine).

**Comment by MattGalvis** [comment]
Yea I was wondering why @robinedwards but if i were to use the change PW call there before to make it work it still gets errors i will put that up in PR now

**Comment by MattGalvis** [comment]
Really unsure how these tests were run locally before @robinedwards do u remember how?

**Comment by MattGalvis** [comment]
@robinedwards sorry I rebased the branch and didn't see the comments you made on a file so I requested for you to take another look.

**Review by trentmurray — CHANGES_REQUESTED** [review]

File: django_neomodel/apps.py [hunk 1]
```diff
@@ -11,10 +11,11 @@ class NeomodelConfig(AppConfig):
     verbose_name = 'Django neomodel'
 
     def read_settings(self):
-        config.DATABASE_URL = getattr(settings, 'NEOMODEL_NEO4J_BOLT_URL', config.DATABASE_URL)
+        # config.DATABASE_URL = getattr(settings, 'NEOMODEL_NEO4J_BOLT_URL', config.DATABASE_URL)
+        config.DATABASE_URL = "bolt://neo4j:0000@localhost:7687"
```
[the remark this answers is not in the document]
Comment (MattGalvis) [reply django_neomodel/apps.py:15]: Yes this was just to run locally

**Comment by vesnikos** [comment]
any updates on this? The updated read_settings in this PR seems to be the correct ones vs the existing version

**Comment by MattGalvis** [comment]
Nope @vesnikos still waiting on review from @robinedwards

File: .gitignore [hunk 2]
```diff
@@ -14,4 +14,6 @@ development.env
 .python-version
 .ropeproject
 \#*\#
-tests/test.db
\ No newline at end of file
+tests/test.db
+.DS_Store
+test.db
```
Comment (robinedwards) [inline .gitignore:19]: Do we need test.db in here twice? DS_Store also?

File: setup.py [hunk 3]
```diff
@@ -12,18 +12,17 @@
     license='MIT',
     packages=find_packages(exclude=('tests',)),
     keywords='neo4j django plugin neomodel',
-    install_requires=['neomodel>=3.2.5', 'django>=1.9'],
+    install_requires=['neomodel>=4.0.1', 'neo4j-driver==4.1.1', 'pytz>=2020.1', 'django==2.2'],
```
Comment (robinedwards) [inline setup.py:15]: Can't we set `django>=2.2` or does a later version brake it? neomodel pulls in neo4j-driver right so no need to include that here

File: setup.py [hunk 4]
```diff
@@ -12,18 +12,17 @@
...
         'License :: OSI Approved :: MIT License',
         'Programming Language :: Python',
         'Topic :: Software Development :: Libraries :: Python Modules',
-        "Programming Language :: Python :: 2.7",
-        "Programming Language :: Python :: 3.3",
-        "Programming Language :: Python :: 3.4",
-        "Programming Language :: Python :: 3.5",
         "Programming Language :: Python :: 3.6",
+        "Programming Language :: Python :: 3.7",
+        # "Programming Language :: Python :: 3.8",
```
Comment (robinedwards) [inline setup.py:26]: Any reason we can't support up to 3.9?

**Review by robinedwards — CHANGES_REQUESTED** [review]

**Comment by robinedwards** [comment]
Sorry for the delay. I've just submitted the review. I think there are a
few things to tidy up before it can be merged along with fixing the travis
tests.

File: setup.py (hunk 3 above)
Comment (MattGalvis) [reply setup.py:15]: thanks

File: setup.py (hunk 4 above)
Comment (MattGalvis) [reply setup.py:26]: in next PR u will see in comment

File: dev [hunk 5]
```diff
@@ -0,0 +1 @@
+python3 -m venv venv; source venv/bin/activate; python setup.py develop;
```
[the remark this answers is not in the document]
Comment (MattGalvis) [reply dev:1]: I've added this file to make contributing easy i will be updating docs to have a Contribute/testing "how to"

File: Changelog [hunk 6]
```diff
@@ -9,7 +9,7 @@ Version 0.0.3 2017-02-19
 Version 0.0.2 2017-02-07
  * Requires neo4j_driver 1.1.0 (Added in neomodel 3.2.0)
  * Cleaned up signals code
- * Add new config options MAX_POOL_SIZE and ENCRYPTED_CONNECTION
+ * Add new config options MAX_CONNECTION_POOL_SIZE and ENCRYPTED_CONNECTION
```
[the remark this answers is not in the document]
Comment (MattGalvis) [reply Changelog:12]: yea your right

File: tests/settings.py [hunk 7]
```diff
@@ -25,9 +24,12 @@
         'PORT': '',
     },
 }
-
-NEOMODEL_NEO4J_BOLT_URL = os.environ.get('NEO4J_BOLT_URL', 'bolt://neo4j:neo4j@localhost:7687')
+DEFAULT_INDEX_TABLESPACE = "10"
```
Comment (robinedwards) [inline tests/settings.py:27]: Sorry whats this?

File: Changelog [hunk 8]
```diff
@@ -1,3 +1,6 @@
+Version 0.0.5 2020-11-20
+ * TODO
```
Comment (robinedwards) [inline Changelog:2]: Can we expand on this?

File: README.rst [hunk 9]
```diff
@@ -151,9 +150,40 @@ Delete all nodes in your database, warning there is no confirmation!
...
 
 .. image:: https://badges.gitter.im/Join%20Chat.svg
    :alt: Join the chat at https://gitter.im/robinedwards/neomodel
    :target: https://gitter.im/robinedwards/neomodel?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge
+
+
+To Contribute
+===================
+
+Video on how to run tests:
```
Comment (robinedwards) [inline README.rst:164]: If there is no link yet can we remove this text?

File: dev [hunk 10]
```diff
@@ -0,0 +1 @@
+python3 -m venv venv; source venv/bin/activate; python setup.py develop; export DJANGO_SETTINGS_MODULE=tests.settings;
```
Comment (robinedwards) [inline dev:1]: I think its better just to put this in the docs as opposed to having a mysterious file people source

**Review by robinedwards — COMMENTED** [review]
A few comments need addressing, the travis build doesn't pass yet but looking good

**Comment by MattGalvis** [comment]
@robinedwards should be good to merge now

**Comment by robinedwards** [comment]
Super thank you @MattGalvis for your hard work :-)

**Pull request merged**